src/vs/workbench/contrib/terminal/browser/terminal.ts

```diff
@@ -262,7 +262,7 @@ export interface ITerminalService extends ITerminalInstanceHost {
 	readonly onDidRequestStartExtensionTerminal: Event<IStartExtensionTerminalRequest>;
 	readonly onDidRegisterProcessSupport: Event<void>;
 	readonly onDidChangeConnectionState: Event<void>;
-
+	readonly onAnyInstanceShellTypeChanged: Event<ITerminalInstance>; // Here or on proposed api? - also there is onDidChangeShellType that already exists..
```

Comment:
`onAnyInstance` events are helpers we can add when needed, this is a good thing but we should move it into the `// Multiplexed events` section